doc-changelog/parse_pr_title.py

```diff
@@ -0,0 +1,204 @@
+            print("Pull request title is blank")
+            exit(1)
+
+    if first_letter.islower():
+        save_env_variable("FIRST_LETTER", "lowercase")
+    else:
+        save_env_variable("FIRST_LETTER", "uppercase")
+
+
+def get_conventional_commit_type(pr_title):
+    pr_title = f"""{pr_title}"""
+    colon_index = pr_title.index(":")
+    cc_type = '"' + pr_title[:colon_index] + '"'
+    save_env_variable("CC_TYPE", cc_type)
+
+
+def changelog_category_cc(cc_type):
+    # Get conventional commit type from env variable
+    cc_type = cc_type.lower()
+
+    print(cc_type)
+
+    cc_type_changelog_dict = {
+        "feat": "added",
+        "fix": "fixed",
+        "docs": "documentation",
+        "build": "dependencies",
+        "revert": "miscellaneous",
+        "style": "miscellaneous",
+        "refactor": "miscellaneous",
+        "perf": "miscellaneous",
+        "test": "test",
+        "chore": "maintenance",
+        "ci": "maintenance",
+    }
+
+    changelog_section = cc_type_changelog_dict[cc_type]
+
+    save_env_variable("CHANGELOG_SECTION", changelog_section)
+
+
+def changelog_cateogry_labels(labels):
+    # Create a list of labels found in the pull request
+    # For example, "enhancement maintenance".split() -> ["enhancement", "maintenance"]
+    existing_labels = labels.split()
+
+    # Dictionary with the key as a label from .github/workflows/label.yml and
+    # value as the corresponding section in the changelog
+    pr_labels = {
+        "enhancement": "added",
+        "bug": "fixed",
+        "documentation": "documentation",
+        "testing": "test",
+        "dependencies": "dependencies",
+        "CI/CD": "maintenance",
+        "maintenance": "maintenance",
+    }
+
+    def get_changelog_section(pr_labels, existing_labels):
+        """Find the changelog section corresponding to the label in the PR."""
+        label_type = ""
+
+        for key, value in pr_labels.items():
+            if key in existing_labels:
+                label_type = value
+                return label_type
+
+        # If no labels are in the PR, it goes into the miscellaneous category
+        label_type = "miscellaneous"
+        return label_type
+
+    save_env_variable(
+        "CHANGELOG_SECTION", get_changelog_section(pr_labels, existing_labels)
+    )
+
+
+def clean_pr_title(pr_title, use_labels):
+    # Retrieve title
+    clean_title = pr_title
+
+    # If not using label, remove conventional commit type from title
+    if use_labels == "False":
+        colon_index = clean_title.index(":")
+        clean_title = clean_title[colon_index + 1 :]
+
+    print(clean_title)
+
+    # Remove extra whitespace
+    clean_title = clean_title.strip()
+
+    # Add backslash in front of backtick and double quote
+    clean_title = clean_title.replace("`", "\\`").replace('"', '\\"')
+
+    save_env_variable("CLEAN_TITLE", clean_title)
+
+
+def add_towncrier_config(org_name, repo_name, web_release_notes):
```

Comment:
@RobPasMue, is this what you were thinking for the template? The web_release_notes is a bool whether or not the person is using or wants to use the release note configuration in documentation vs. the CHANGELOG.md file. If the [tool.towncrier] section doesn't exist, it'll add it based on the web_release_notes bool

Reply:
Yep exactly but you could simplify it. Just accept a boolean to either include the entire towncrier section or not. You don't have to do all the conditional checks I believe. Either you add the entire section or you don't add it based on the users input ``use_pyansys_towncrier_default_config`` or something like that